src/gardenlinux/features/cname_main.py

```diff
@@ -44,19 +44,19 @@ def main():
 
     assert re_match, f"Not a valid GardenLinux canonical name {args.cname}"
 
-    arch = None
+    arch = args.arch
     gardenlinux_root = dirname(args.feature_dir)
     version = args.version
 
-    if args.arch is not None:
-        arch = args.arch
-
-    if args.version is not None:
-        version = args.version
+    if gardenlinux_root == "":
+        gardenlinux_root = "."
 
     if not version:
-        version_data = get_version_and_commit_id_from_files(gardenlinux_root)
-        version = f"{version_data[0]}-{version_data[1]}"
+        try:
+            version_data = get_version_and_commit_id_from_files(gardenlinux_root)
+            version = f"{version_data[0]}-{version_data[1]}"
+        except RuntimeError:
+            pass  # Final failing of version parsing is handled in cname below
```

Comment:
Still a bit uneasy that we pass this error silently.
Personally would prefer an early failure approach to stuff like this and `raise RuntimeError`.

Alternatively, logging the error as a `warning` would also be an option, if this failure is not critical. Anyway it would probably help us debugging in the future.
```suggestion
        try:
            version_data = get_version_and_commit_id_from_files(gardenlinux_root)
            version = f"{version_data[0]}-{version_data[1]}"
        except RuntimeError:
            logging.warning(f"Warning: Failed to parse version and commit ID from files: {e}")
            raise
```

Reply:
I personally have the feeling we over engineer this aspect in `__main__` as an error will be presented to the user anyway. That being said I do agree with you that clean code aspects are important as well and debugging would be easier. I added a message even thought `logging` will not be configured at this stage.